plugins/router/template/router.go

```diff
@@ -252,6 +280,15 @@ func (r *templateRouter) AddEndpoints(id string, endpoints []Endpoint) {
 	r.state[id] = frontend
 }
 
+// cleanUpServiceAliasConfig performs any necessary steps to clean up a service alias config before deleting it from
+// the router.  Right now the only clean up step is to remove any of the certificates on disk.
+func (r *templateRouter) cleanUpServiceAliasConfig(cfg *ServiceAliasConfig) {
+	err := r.certManager.deleteCertificatesForConfig(cfg)
+	if err != nil {
+		glog.Errorf("Error deleting certificates for route %s, the route will still be deleted but files may remain in the container: %v", cfg.Host, err)
```

Comment:
Hmm, I like it.  Writing to a different directory and doing a swap simplifies this a lot.